project_redss/analysis_file.py

```diff
@@ -97,35 +97,36 @@ def generate(user, data, csv_by_message_output_path, csv_by_individual_output_pa
             if plan.raw_field not in survey_keys:
                 survey_keys.append(plan.raw_field)
 
+        # Convert survey codes to their string values
         for td in data:
             td.append_data(
                 {plan.analysis_file_key: plan.code_scheme.get_code_with_id(td[plan.coded_field]["CodeID"]).string_value
                  for plan in PipelineConfiguration.SURVEY_CODING_PLANS},
                 Metadata(user, Metadata.get_call_location(), time.time())
             )
 
-            td.append_data({
-                "rqa_s01e02_integrate_return":
-                    CodeSchemes.S01E02_INTEGRATE_RETURN.get_code_with_id(
-                        td["rqa_s01e02_integrate_return_coded"]["CodeID"]).string_value
-            }, Metadata(user, Metadata.get_call_location(), time.time()))
-
-            td.append_data({
-                "rqa_s01e03_yes_no":
-                    CodeSchemes.S01E03_YES_NO_AMB.get_code_with_id(
-                        td["rqa_s01e03_yes_no_amb_coded"]["CodeID"]).string_value
-            }, Metadata(user, Metadata.get_call_location(), time.time()))
-
+        # Convert the operator code to its string value
         for td in data:
             td.append_data(
                 {"operator": CodeSchemes.OPERATOR.get_code_with_id(td["operator_coded"]["CodeID"]).string_value},
                 Metadata(user, Metadata.get_call_location(), time.time())
             )
 
-        # Translate keys to final values for analysis
+        # Convert RQA binary codes to their string values
+        for td in data:
+            td.append_data(
+                {plan.binary_analysis_file_key:
+                 plan.binary_code_scheme.get_code_with_id(td[plan.binary_coded_field]["CodeID"]).string_value
+                 for plan in PipelineConfiguration.RQA_CODING_PLANS if plan.binary_code_scheme is not None},
+                Metadata(user, Metadata.get_call_location(), time.time())
+            )
+
+        # Translate the RQA reason codes to matrix values
         matrix_keys = []
 
         for plan in PipelineConfiguration.RQA_CODING_PLANS:
+            # TODO: If this was a list, codes would come out in the order they're in in the schemes, rather than
```

Comment:
I think sorted in the order they're in in the schemes is better than by alpha. E.g. it sorts all of the control codes to the end

Reply:
Will address on ADSS